app/components/candidate_interface/course_choices_review_component.rb

```diff
@@ -268,6 +269,15 @@ def application_predates_visa_sponsorship_information?(application_choice)
       application_choice.application_form.recruitment_cycle_year < 2022
     end
 
+    def course_can_sponsor_visa?(application_choice)
+      (
+        application_choice.course.salary? && application_choice.course.can_sponsor_skilled_worker_visa?
+      ) ||
+        (
+          !application_choice.course.salary? && application_choice.course.can_sponsor_student_visa?
+        )
+    end
```

Comment:
I use a similar logic from the provider_can_sponsor_visa? below. Let me know if you prefer to have only one method that vqries between course and provider (e.g `can_sponsor_visa?(application_choice, :course) ||  can_sponsor_visa?(application_choice, :provider)`. I am happy to change if that's the preferred approach.